qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+    of active blocks that is automatically collapsed upon exiting the context.
+
+    Args:
+        transform: Transform to decorate as context.
+        transform_kwargs: Additional override keyword arguments for the
+            decorated transform.
+
+    Returns:
+        A function that generates a new transformation ``ContextManager``.
+    """
+    def wrap(function):  # pylint: disable=unused-argument
+        @functools.wraps(function)
+        @contextmanager
+        def wrapped_transform(*args, **kwargs):
+            builder = _active_builder()
+            active_block = builder.block
+            transform_block = Schedule()
+            builder.set_active_block(transform_block)
+            try:
+                yield
+            finally:
+                builder._compile_lazy_circuit()
+                transformed_block = transform(transform_block,
+                                              *args,
+                                              **kwargs,
+                                              **transform_kwargs)
+                builder.set_active_block(active_block)
+                builder.append_block(transformed_block)
+        return wrapped_transform
+
+    return wrap
+
+
+@_transform_context(transforms.align_left)
+def align_left() -> ContextManager[None]:
+    """Left alignment pulse scheduling context.
+
+    Pulse instructions within this context are scheduled as early as possible
+    by shifting them left to the earliest available time.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        with pulse.build() as pulse_prog:
+            with pulse.align_left():
+                # this pulse will start at t=0
+                pulse.play(pulse.Constant(100, 1.0), d0)
+                # this pulse will start at t=0
+                pulse.play(pulse.Constant(20, 1.0), d1)
+
+        assert pulse_prog.ch_start_time(d0) == pulse_prog.ch_start_time(d1)
+    """
+
+
+@_transform_context(transforms.align_right)
+def align_right() -> ContextManager[None]:
+    """Right alignment pulse scheduling context.
+
+    Pulse instructions within this context are scheduled as late as possible
+    by shifting them right to the latest available time.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        with pulse.build() as pulse_prog:
+            with pulse.align_right():
+                # this pulse will start at t=0
+                pulse.play(pulse.Constant(100, 1.0), d0)
+                # this pulse will start at t=80
+                pulse.play(pulse.Constant(20, 1.0), d1)
+
+        assert pulse_prog.ch_stop_time(d0) == pulse_prog.ch_stop_time(d1)
+    """
+
+
+@_transform_context(transforms.align_sequential)
+def align_sequential() -> ContextManager[None]:
```

Comment:
I prefer align sequential. This makes it clear this is an alignment context and we can have `align_left`, `align_right`, `align_center`, `align_sequential`. It's clear that these are mutually exclusive options with this syntax. I've thought about this a lot and it seems to be really unclear if we have just `pulse.left()`, or confusing if we have `pulse.align_left` and `pulse.sequential` as this may be viewed as inconsistent.

I'm definitely open to discussion here.